### PR TITLE
Added hp-version for `VectorTools::compute_mean_value()`.

### PR DESCRIPTION
**Pull request by marcfehling**
Quickly written patch to allow `VectorTools::compute_mean_value()` for the hp-case. Still needs a test.

### CODE REVIEW

**Comment by bangerth** [comment]
Your compiler is too new:
```
/Users/runner/work/dealii/dealii/include/deal.II/numerics/vector_tools_mean_value.templates.h:217:35: error: use of class template 'MappingCollection' requires template arguments
    return compute_mean_value(hp::MappingCollection(mapping),
                                  ^
/Users/runner/work/dealii/dealii/include/deal.II/numerics/vector_tools_mean_value.h:33:9: note: template is declared here
  class MappingCollection;
        ^
```

**Review by bangerth — COMMENTED** [review]
Other than the compile error, this looks good. Would you be willing to add a test?

File: include/deal.II/numerics/vector_tools_mean_value.templates.h [hunk 1]
```diff
@@ -216,7 +216,7 @@ namespace VectorTools
   {
     return compute_mean_value(hp::MappingCollection(mapping),
```
Comment (bangerth) [inline include/deal.II/numerics/vector_tools_mean_value.templates.h:217]: You need the same fix with `<dim>` here

**Comment by marcfehling** [comment]
I provided a test based on `tests/mpi/compute_mean_value`.

I opened the PR a bit too early, now it's ready for a final review. My apologies.

**Review by bangerth — APPROVED** [review]

**Pull request merged**